### PR TITLE
Update leak_tracker to use LeakTesting.

### PR DESCRIPTION
**Pull request by polina-c**
Contributes to https://github.com/flutter/flutter/issues/135856

### CODE REVIEW

File: pkgs/leak_tracker/lib/src/leak_tracking/_primitives/model.dart [hunk 1]
```diff
@@ -115,6 +92,25 @@ class IgnoredLeaksSet {
     if (ignoreAll) return true;
     return byClass.containsKey(className) && byClass[className] == null;
   }
+
+  @override
+  bool operator ==(Object other) {
+    if (identical(other, this)) {
+      return true;
+    }
+    if (other.runtimeType != runtimeType) {
```
[the remark this answers is not in the document]
Comment (polina-c) [reply pkgs/leak_tracker/lib/src/leak_tracking/_primitives/model.dart:101]: I am using this boilerplate: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#common-boilerplates-for-operator--and-hashcode

I would leave it as is.

Hopefully, when we gat macros, we will get rid of all this manually crafted code :)

File: pkgs/leak_tracker/lib/src/leak_tracking/_primitives/model.dart [hunk 2]
```diff
@@ -145,6 +141,25 @@ class IgnoredLeaks {
...
+      return false;
+    }
```
[the remark this answers is not in the document]
Comment (polina-c) [reply pkgs/leak_tracker/lib/src/leak_tracking/_primitives/model.dart:155]: it is standard boilerplate

File: pkgs/leak_tracker_flutter_testing/lib/src/leak_testing.dart [hunk 3]
```diff
@@ -183,4 +186,28 @@ class LeakTesting {
...
+  final MemoryBaselining baselining;
+
+  @override
+  bool operator ==(Object other) {
+    if (identical(other, this)) {
+      return true;
+    }
+    if (other.runtimeType != runtimeType) {
+      return false;
+    }
```
[the remark this answers is not in the document]
Comment (polina-c) [reply pkgs/leak_tracker_flutter_testing/lib/src/leak_testing.dart:203]: it is standard boilerplate

File: pkgs/leak_tracker/lib/src/leak_tracking/_primitives/model.dart [hunk 4]
```diff
@@ -263,33 +279,28 @@ class LeakTrackingConfig {
...
-    this.isLeakTrackingPaused = false,
+    this.ignoredLeaks = const IgnoredLeaks(),
+    this.ignoreLeaks = false,
     this.name,
     this.leakDiagnosticConfig = const LeakDiagnosticConfig(),
     this.baselining,
   });
 
-  const PhaseSettings.paused() : this(isLeakTrackingPaused: true);
+  const PhaseSettings.paused() : this(ignoreLeaks: true);
```
Comment (kenzieschmoll) [inline pkgs/leak_tracker/lib/src/leak_tracking/_primitives/model.dart:297]: nit: rename paused to ignored

**Review by kenzieschmoll — APPROVED** [review]
lgtm with one more comment

**Pull request merged**